src/components/WeeklyCalendar/week/styled/event.js

```diff
@@ -6,8 +6,12 @@ const StyledEvent = attachThemeAttrs(styled.div)`
     text-align: left;
     font-size: 12px;
     border: 1px solid ${props => replaceAlpha(props.palette.background.main, 0.8)};
-    color: ${props => props.palette.getContrastText(props.palette.brand.main)};
-    background-color: ${props => props.palette.brand.main};
+    color: ${props =>
+        props.color ||
+        (props.backgroundColor
+            ? props.palette.getContrastText(props.backgroundColor)
+            : props.palette.getContrastText(props.palette.brand.main))};
+    background-color: ${props => props.backgroundColor || props.palette.brand.main};
```

Comment:
here we need to check the backgroundColor and color props passed are valid colors, here `src/styles/helpers/color/isValidColor.js` we have a helper for this

Reply:
this idea crossed my mind, but i didn't know about the existence of this function until now